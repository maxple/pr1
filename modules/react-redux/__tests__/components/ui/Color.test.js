import Color from '../../../components/ui/Color'
import {mount, render, shallow} from 'enzyme'

describe('Color UI Component', () => {
    it('Renders correct properties shallow', () =>
        expect(shallow(
            <Color title="Test Color"
                   color="#F0F0F0"
                   rating={3}
                   timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />))
            .toMatchSnapshot(),
    )

    it('Renders correct properties mount', () =>
        expect(mount(
            <Color title="Test Color"
                   color="#F0F0F0"
                   rating={3}
                   timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />))
            .toMatchSnapshot(),
    )

    it('Renders correct properties render', () =>
        expect(render(
            <Color title="Test Color"
                   color="#F0F0F0"
                   rating={3}
                   timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />))
            .toMatchSnapshot(),
    )
})
