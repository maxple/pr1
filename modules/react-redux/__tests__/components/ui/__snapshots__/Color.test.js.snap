// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color UI Component Renders correct properties mount 1`] = `
<Color
  color="#F0F0F0"
  onRate={[Function]}
  onRemove={[Function]}
  rating={3}
  timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
  title="Test Color"
>
  <section
    className="color"
  >
    <h1>
      Test Color
    </h1>
    <button
      onClick={[Function]}
    >
      <Component>
        <IconBase
          attr={
            Object {
              "viewBox": "0 0 448 512",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
              key="0"
            />
          </svg>
        </IconBase>
      </Component>
    </button>
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#F0F0F0",
        }
      }
    />
    <TimeAgo
      timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
    >
      <div
        className="time-ago"
      >
        4/11/2016
      </div>
    </TimeAgo>
    <div>
      <StarRating
        onRate={[Function]}
        starsSelected={3}
      >
        <div
          className="star-rating"
        >
          <Star
            key="0"
            onClick={[Function]}
            selected={true}
          >
            <div
              className="star selected"
              onClick={[Function]}
            />
          </Star>
          <Star
            key="1"
            onClick={[Function]}
            selected={true}
          >
            <div
              className="star selected"
              onClick={[Function]}
            />
          </Star>
          <Star
            key="2"
            onClick={[Function]}
            selected={true}
          >
            <div
              className="star selected"
              onClick={[Function]}
            />
          </Star>
          <Star
            key="3"
            onClick={[Function]}
            selected={false}
          >
            <div
              className="star"
              onClick={[Function]}
            />
          </Star>
          <Star
            key="4"
            onClick={[Function]}
            selected={false}
          >
            <div
              className="star"
              onClick={[Function]}
            />
          </Star>
          <p>
            3
             of 
            5
             stars
          </p>
        </div>
      </StarRating>
    </div>
  </section>
</Color>
`;

exports[`Color UI Component Renders correct properties render 1`] = `
<section
  class="color"
>
  <h1>
    Test Color
  </h1>
  <button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
      />
    </svg>
  </button>
  <div
    class="color"
    style="background-color:#F0F0F0"
  />
  <div
    class="time-ago"
  >
    4/11/2016
  </div>
  <div>
    <div
      class="star-rating"
    >
      <div
        class="star selected"
      />
      <div
        class="star selected"
      />
      <div
        class="star selected"
      />
      <div
        class="star"
      />
      <div
        class="star"
      />
      <p>
        3 of 5 stars
      </p>
    </div>
  </div>
</section>
`;

exports[`Color UI Component Renders correct properties shallow 1`] = `
<section
  className="color"
>
  <h1>
    Test Color
  </h1>
  <button
    onClick={[Function]}
  >
    <Component />
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    }
  />
  <TimeAgo
    timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
  />
  <div>
    <StarRating
      onRate={[Function]}
      starsSelected={3}
    />
  </div>
</section>
`;
